### PR TITLE
 Math Operation

### PR DESCRIPTION
**Pull request by VasiliZas**
Math Operation Task

### CODE REVIEW

**Review by PavelBoPavel — CHANGES_REQUESTED** [review]

File: homework/Vasili_Zasinets/src/main/java/com/tutrit/java/quickstart/homework/mathoperation/MathOperationsDemo.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
+package com.tutrit.java.quickstart.homework.mathoperation;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class MathOperationsDemo {
+    public static void main(String[] args) {
+
+        Logger log = LoggerFactory.getLogger("MathOperations ");
```
[the remark this answers is not in the document]
Comment (VasiliZas) [reply homework/Vasili_Zasinets/src/main/java/com/tutrit/java/quickstart/homework/mathoperation/MathOperationsDemo.java]: fixed

**Review by Puzzletania — APPROVED** [review]

**Review by trump-ny — DISMISSED** [review]

File: homework/Vasili_Zasinets/src/main/java/com/tutrit/java/quickstart/homework/mathoperation/MathOperationsDemo.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
+package com.tutrit.java.quickstart.homework.mathoperation;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class MathOperationsDemo {
+
+    Logger log = LoggerFactory.getLogger("MathOperations ");
+
+    int sum(int a, int b) {
```
Comment (KaydunovDenis) [inline homework/Vasili_Zasinets/src/main/java/com/tutrit/java/quickstart/homework/mathoperation/MathOperationsDemo.java:10]: do it as public methods

**Review by KaydunovDenis — APPROVED** [review]

**Review by PavelBoPavel — DISMISSED** [review]

File: homework/Vasili_Zasinets/src/main/java/com/tutrit/java/quickstart/homework/mathoperation/MathOperationsDemo.java (hunk 2 above)
Comment (KaydunovDenis) [reply homework/Vasili_Zasinets/src/main/java/com/tutrit/java/quickstart/homework/mathoperation/MathOperationsDemo.java:10]: Best practice explicitly specify the access modifier for methods

**Review by KaydunovDenis — CHANGES_REQUESTED** [review]

File: homework/Vasili_Zasinets/src/main/java/com/tutrit/java/quickstart/homework/mathoperation/MathOperationsDemo.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    }
+
+    int subtraction(int a, int b) {
+        log.info("Subtraction {}", a - b);
+        return a - b;
+    }
+
+    float division(int a, int b) {
+        log.info("Division {}", a / b);
+        return (float) a / b;
```
Comment (max-java) [inline homework/Vasili_Zasinets/src/main/java/com/tutrit/java/quickstart/homework/mathoperation/MathOperationsDemo.java:23]: maybe double works better? 
also, it's less error prone if you log and return the same var, like 
```suggestion
    float division(int a, int b) {
        float result = (float) a / b
        log.info("Division {}", result);
        return result;
```
Also, in this case you don't need to perform division twice.

File: homework/Vasili_Zasinets/src/main/java/com/tutrit/java/quickstart/homework/mathoperation/RandomGeneratorDemo.java [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+        Logger log = LoggerFactory.getLogger("MathRandom");
+        float a = (float) (Math.random() * 10);
+        log.info("Random a {}", a);
+        float b = (float) (Math.random() * 20);
+        log.info("Random b {}", b);
+        float c = (float) (Math.random() * 33);
+        log.info("Random c {}", c);
+        float sum = a + b + c;
+        log.info("Sum a+b+c {}", sum);
```
Comment (max-java) [inline homework/Vasili_Zasinets/src/main/java/com/tutrit/java/quickstart/homework/mathoperation/RandomGeneratorDemo.java:17]: Better move this to test and test it

**Review by max-java — CHANGES_REQUESTED** [review]
use quickstart project, perform sonar verification and see minor comments

File: homework/Vasili_Zasinets/src/main/java/com/tutrit/java/quickstart/homework/mathoperation/MathOperationsDemo.java (hunk 3 above)
Comment (VasiliZas) [reply homework/Vasili_Zasinets/src/main/java/com/tutrit/java/quickstart/homework/mathoperation/MathOperationsDemo.java:23]: fixed

File: homework/Vasili_Zasinets/src/main/java/com/tutrit/java/quickstart/homework/mathoperation/MathOperationsDemo.java (hunk 2 above)
Comment (VasiliZas) [reply homework/Vasili_Zasinets/src/main/java/com/tutrit/java/quickstart/homework/mathoperation/MathOperationsDemo.java:10]: fixed

**Review by KaydunovDenis — DISMISSED** [review]

File: homework/Vasili_Zasinets/MathOperationTask/src/test/java/com/tutrit/java/quickstart/homework/mathoperation/RandomGeneratorDemoTest.java [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+    RandomGeneratorDemo randomGeneratorDemo = new RandomGeneratorDemo();
+
+    @Test
+    public void randomGenerator() {
+        Logger log = LoggerFactory.getLogger("MathRandom");
+        log.info("Random a {}", randomGeneratorDemo.randomGeneratorA());
+        log.info("Random b {}", randomGeneratorDemo.randomGeneratorB());
+        log.info("Random c {}", randomGeneratorDemo.randomGeneratorC());
+        log.info("Sum a+b+c {}", randomGeneratorDemo.SumRandom());
+    }
```
Comment (KaydunovDenis) [inline homework/Vasili_Zasinets/MathOperationTask/src/test/java/com/tutrit/java/quickstart/homework/mathoperation/RandomGeneratorDemoTest.java:17]: What are you testing here? Your test return "true" always.

**Review by KaydunovDenis — CHANGES_REQUESTED** [review]

File: homework/Vasili_Zasinets/MathOperationTask/src/test/java/com/tutrit/java/quickstart/homework/mathoperation/RandomGeneratorDemoTest.java (hunk 5 above)
Comment (VasiliZas) [reply homework/Vasili_Zasinets/MathOperationTask/src/test/java/com/tutrit/java/quickstart/homework/mathoperation/RandomGeneratorDemoTest.java:17]: "max-java 3 days ago Collaborator
Better move this to test and test it"

**Review by KaydunovDenis — APPROVED** [review]

**Review by max-java — APPROVED** [review]
lgfm

**Pull request merged**